src/DOCINewtonOrbitalOptimizer.cpp

```diff
@@ -159,7 +159,7 @@ void DOCINewtonOrbitalOptimizer::solve(BaseSolverOptions& solver_options, const
 
 
         // Transform the integrals to the new orthonormal basis
-        this->ham_par.rotate(U);  // this checks if U is actually unitary
+        ham_par.rotate(SquareMatrix<double>(U));  // this checks if U is actually unitary
```

Comment:
is `SquareMatrix<double>` required?

Reply:
Yes, I'm getting
> candidate template ignored: could not match 'SquareMatrix' against 'Matrix'
when compiling with AppleClang. However, this syntax should be fixed in #238.